src/users.ts

```diff
@@ -4,76 +4,77 @@ import { APIClient } from '@textile/users-grpc/users_pb_service'
 import { ServiceError } from '@textile/hub-grpc/hub_pb_service'
 import { Context } from '@textile/context'
 import { ThreadID } from '@textile/threads-id'
+import { Client } from '@textile/threads-client'
 
 const logger = log.getLogger('users')
 
-/**
- * Buckets is a web-gRPC wrapper client for communicating with the web-gRPC enabled Textile Buckets API.
- */
-export class Users {
-  private client: APIClient
-
-  /**
-   * Creates a new gRPC client instance for accessing the Textile Buckets API.
-   * @param context The context to use for interacting with the APIs. Can be modified later.
-   */
-  constructor(public context: Context = new Context()) {
-    this.client = new APIClient(context.host, {
-      transport: context.transport,
-      debug: context.debug,
-    })
+// Module augmentation to add methods to Peer from core
+declare module '@textile/threads-client' {
+  interface Client {
+    getThread(name: string, ctx?: Context): Promise<pb.GetThreadReply.AsObject>
+    listThreads(ctx?: Context): Promise<pb.ListThreadsReply.AsObject>
   }
+}
 
-  /**
-   * Returns a Thread by name.
-   * @param name The name of the Thread.
-   * @param ctx Context containing gRPC headers and settings.
-   * These will be merged with any internal credentials.
-   */
-  async getThread(name: string, ctx?: Context) {
-    logger.debug('get thread request')
-    return new Promise<pb.GetThreadReply.AsObject>((resolve, reject) => {
-      const req = new pb.GetThreadRequest()
-      req.setName(name)
-      this.client.getThread(
-        req,
-        this.context.withContext(ctx).toMetadata(),
-        (err: ServiceError | null, message: pb.GetThreadReply | null) => {
-          if (err) reject(err)
-          const msg = message?.toObject()
-          if (msg) {
-            msg.id = ThreadID.fromBytes(Buffer.from(msg.id as string, 'base64')).toString()
-          }
-          resolve(msg)
-        },
-      )
-    })
-  }
+/**
+ * Returns a Thread by name.
+ * @param name The name of the Thread.
+ * @param ctx Context containing gRPC headers and settings.
+ * These will be merged with any internal credentials.
+ */
+Client.prototype.getThread = async function (name: string, ctx?: Context) {
+  logger.debug('get thread request')
+  const client = new APIClient(this.serviceHost, {
```

Comment:
Because we're 'piggy backing' off of threads client, we create out own client here each time, but use all the options and settings from threads client.